snap/snapcraft.yaml

```diff
@@ -0,0 +1,118 @@
+name: edgex-device-onvif-camera
+base: core20
+license: Apache-2.0
+adopt-info: metadata
+
+architectures:
+  - build-on: amd64
+  - build-on: arm64
+
+grade: stable
+confinement: strict
+
+slots:
+  edgex-secretstore-token:
+    interface: content
+    source:
+      write: [$SNAP_DATA/device-onvif-camera]
+
+plugs:
+  device-config:
+    interface: content
+    target: $SNAP_DATA/config/device-onvif-camera
+
+apps:
+  device-onvif-camera:
+    adapter: full
+    command: bin/device-onvif-camera -confdir $SNAP_DATA/config/device-onvif-camera -profile res --registry $CONSUL_ADDR
+    command-chain:
+      - bin/source-env-file.sh
+    environment:
+      CONSUL_ADDR: "consul://localhost:8500"
+      APPCUSTOM_PROVISIONWATCHERDIR: $SNAP_DATA/config/device-onvif-camera/res/provision_watchers
+      DEVICE_PROFILESDIR: $SNAP_DATA/config/device-onvif-camera/res/profiles
+      DEVICE_DEVICESDIR: $SNAP_DATA/config/device-onvif-camera/res/devices
+      SECRETSTORE_SECRETSFILE: $SNAP_DATA/device-onvif-camera/onvif-credentials.json
```

Comment:
Please investigate why `Writable.InsecureSecrets.Camera001` inside the server configuration file isn't sufficient.

Reply:
Thanks for your suggestion. I have removed `SECRETSTORE_SECRETSFILE` and `SECRETSTORE_DISABLESCRUBSECRETSFILE`.

I am getting `Unable to retrieve camera credentials from SecretProvider at path 'credentials001'` error now after setting up credentials in configuration.toml file. Here is the full log: https://pastebin.ubuntu.com/p/ff3knnx7cX/

Reply:
Based on this comment, that section of configuration only applies when secret store is not used.

https://github.com/edgexfoundry/device-onvif-camera/blob/e5053f8c731e365d1df3f2055d168c1a15fdfcf5/cmd/res/configuration.toml#L3

Reply:
With that said, I think we still shouldn't set this env var here because the user may wanna pre-populate the secret store via other means, e.g. using Vault's API.

Read more here: https://docs.edgexfoundry.org/2.3/security/SeedingServiceSecrets/

If the user wants to provide the secrets locally, or disable the scrubbing (enabled by default), it can be done on demand using snap options.